book/src/pong_tutorial/pong_tutorial_03.md

```diff
@@ -0,0 +1,232 @@
+# Moving the paddles
+
+This is a big step. After this we'll be able to move our entities around, using 
+the keyboard. So, basically we'll have a video game. The good news is the 
+structure we have in place so far makes adding this functionality relatively 
+straightforward.
+
+First let's make a new file called `paddle.rs` which will hold our 
+`PaddleSystem` struct and implementation. Remember how ECS stands for "Entity-
+component system"? Well, this is the "system" part. Systems iterate through 
+components and *do stuff*.
+
+Let's import what we'll need and declare our `PaddleSystem` struct:
+
+```rust,ignore
+use pong::{Paddle, Side};
+
+use amethyst::core::timing::Time;
+use amethyst::core::transform::LocalTransform;
+use amethyst::ecs::{Fetch, Join, System, WriteStorage};
+use amethyst::input::InputHandler;
+
+pub struct PaddleSystem;
+```
+
+Now we're going to implement the `System` trait for `PaddleSystem`. This 
+involves the Rust concept of lifetimes, which you might have managed to avoid in 
+your Rust learning journey thus far. Don't be scared! 
+[Here's the relevant part of the The Rust Book][lifetimes], if you want a 
+refresher. But basically, all we're doing is saying this method has a lifetime 
+`s`, and all the data this method relies on needs to live as long as `s`. This 
+is how we can run multiple systems in parallel without fear.
+
+```rust,ignore
+impl<'s> System<'s> for PaddleSystem {
+    //We'll define the system in here
+}
+```
+
+First off we need to define the `SystemData`, which is a tuple of items which 
+each implement `SystemData`, such as `WriteStorage`, `ReadStorage`, and `Fetch`. 
+
+```rust,ignore
+type SystemData = (
+    WriteStorage<'s, Paddle>,
+    WriteStorage<'s, LocalTransform>,
+    Fetch<'s, Time>,
+    Fetch<'s, InputHandler<String, String>>,
+);
+```
+
+Now we're done worrying about liftimes! Painless, right? Also, you might notice 
+that this `SystemData` tuple is a nice and succinct description of what we want 
+to do with this `System`: we want to change the `LocalTransform` of a `Paddle` 
+over `Time` based on keyboard `Input`.
+
+But we don't have `Time` or an `InputHandler` yet. Let's fix that. 
+
+In `pong.rs` we'll import `Time`:
+
+```rust,ignore
+use amethyst::core::timing::Time;
+```
+
+And add it as a resource to our `on_start` method.
+
+```rust,ignore
+fn on_start(&mut self, world: &mut World) {
+    world.add_resource(Time::default()); //Add this line
```

Comment:
This is redundant.  The engine does this automatically.  I don't know why this was written into the pong example in the first place, so we should fix it there too.

Reply:
Good to know! Fixed.